src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEModuleSymbol.cs

```diff
@@ -641,32 +642,48 @@ internal NamedTypeSymbol LookupTopLevelMetadataType(ref MetadataTypeName emitted
         }
 
         /// <summary>
-        /// If this module forwards the given type to another assembly, return that assembly;
-        /// otherwise, return null.
+        /// Returns a list of the assemblies this module forwards the given type to.
         /// </summary>
         /// <param name="fullName">Type to look up.</param>
-        /// <returns>Assembly symbol or null.</returns>
+        /// <returns>An <see cref="ImmutableArray"/> of the forwarded to assemblies.</returns>
         /// <remarks>
-        /// The returned assembly may also forward the type.
+        /// The returned assemblies may also forward the type.
         /// </remarks>
-        internal AssemblySymbol GetAssemblyForForwardedType(ref MetadataTypeName fullName)
+        internal ImmutableArray<AssemblySymbol> GetAssembliesForForwardedType(ref MetadataTypeName fullName)
         {
             string matchedName;
-            AssemblyReferenceHandle assemblyRef = Module.GetAssemblyForForwardedType(fullName.FullName, ignoreCase: false, matchedName: out matchedName);
-            return assemblyRef.IsNil ? null : this.GetReferencedAssemblySymbol(assemblyRef);
+            HashSet<AssemblyReferenceHandle> assemblyRefs = Module.GetAssemblyRefsForForwardedType(fullName.FullName, ignoreCase: false, matchedName: out matchedName);
+            var assemblies = ArrayBuilder<AssemblySymbol>.GetInstance();
+
+            if (assemblyRefs != null)
+            {
+                foreach (var reference in assemblyRefs)
+                {
+                    var assembly = this.GetReferencedAssemblySymbol(reference);
+                    if (assembly != null)
```

Comment:
So we have two options in case of invalid metadata, either discard it, or keep the null references. I chose to discard invalid values for now. If we have a repro for invalid IL that we need to handle in a different way, I'll open another bug for that. But for now, I chose to discard it.